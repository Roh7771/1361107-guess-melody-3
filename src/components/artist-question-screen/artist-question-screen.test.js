import React from "react";
import renderer from "react-test-renderer";
import ArtistQuestionScreen from "./artist-question-screen";

const AVATAR_URL = `https://api.adorable.io/avatars/128`;
let id = 1;
const question = {
  type: `artist`,
  song: {
    artist: `Bob Marley`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`
  },
  answers: [
    {
      picture: `${AVATAR_URL}/${Math.random()}`,
      artist: `John Snow`,
      id: id++
    },
    {
      picture: `${AVATAR_URL}/${Math.random()}`,
      artist: `Colin Boston`,
      id: id++
    },
    {
      picture: `${AVATAR_URL}/${Math.random()}`,
      artist: `Tom Hardy`,
      id: id++
    }
  ]
};

it(`Should ArtistQuestionScreen render correctly`, () => {
  const tree = renderer
    .create(<ArtistQuestionScreen question={question} onAnswer={() => {}} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
