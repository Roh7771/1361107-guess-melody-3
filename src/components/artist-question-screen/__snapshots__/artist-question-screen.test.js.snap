// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ArtistQuestionScreen render correctly 1`] = `
<main
  className="app"
>
  <svg
    style={
      Object {
        "left": "-1200em",
        "position": "absolute",
      }
    }
    xmlns="http://www.w3.org/2000/svg"
  >
    <filter
      id="blur"
    >
      <feGaussianBlur
        in="SourceGraphic"
        stdDeviation="5"
      />
      <feOffset
        dx="0"
        dy="0"
      />
      <feMerge>
        <feMergeNode />
        <feMergeNode
          in="SourceGraphic"
        />
      </feMerge>
    </filter>
  </svg>
  <section
    className="main"
    id="root"
  >
    <section
      className="game game--artist"
    >
      <header
        className="game__header"
      >
        <a
          className="game__back"
          href="#"
        >
          <span
            className="visually-hidden"
          >
            Сыграть ещё раз
          </span>
          <img
            alt="Угадай мелодию"
            className="game__logo"
            src="img/melody-logo-ginger.png"
          />
        </a>
        <svg
          className="timer"
          viewBox="0 0 780 780"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            className="timer__line"
            cx="390"
            cy="390"
            r="370"
            style={
              Object {
                "filter": "url(#blur)",
                "transform": "rotate(-90deg) scaleY(-1)",
                "transformOrigin": "center",
              }
            }
          />
        </svg>
        <div
          className="game__mistakes"
        >
          <div
            className="wrong"
          />
          <div
            className="wrong"
          />
          <div
            className="wrong"
          />
        </div>
      </header>
      <section
        className="game__screen"
      >
        <h2
          className="game__title"
        >
          Кто исполняет эту песню?
        </h2>
        <div
          className="game__track"
        >
          <div
            className="track"
          >
            <button
              className="track__button track__button--play"
              type="button"
            />
            <div
              className="track__status"
            >
              <audio
                src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
              />
            </div>
          </div>
        </div>
        <form
          className="game__artist"
        >
          <div
            className="artist"
          >
            <input
              className="artist__input visually-hidden"
              id="answer-0"
              name="answer"
              onChange={[Function]}
              type="radio"
              value="answer-0"
            />
            <label
              className="artist__name"
              htmlFor="answer-0"
            >
              <img
                alt="John Snow"
                className="artist__picture"
                src="https://api.adorable.io/avatars/128/0.4038616055791029"
              />
              John Snow
            </label>
          </div>
          <div
            className="artist"
          >
            <input
              className="artist__input visually-hidden"
              id="answer-1"
              name="answer"
              onChange={[Function]}
              type="radio"
              value="answer-1"
            />
            <label
              className="artist__name"
              htmlFor="answer-1"
            >
              <img
                alt="Colin Boston"
                className="artist__picture"
                src="https://api.adorable.io/avatars/128/0.6094787075450023"
              />
              Colin Boston
            </label>
          </div>
          <div
            className="artist"
          >
            <input
              className="artist__input visually-hidden"
              id="answer-2"
              name="answer"
              onChange={[Function]}
              type="radio"
              value="answer-2"
            />
            <label
              className="artist__name"
              htmlFor="answer-2"
            >
              <img
                alt="Tom Hardy"
                className="artist__picture"
                src="https://api.adorable.io/avatars/128/0.4316569233682972"
              />
              Tom Hardy
            </label>
          </div>
        </form>
      </section>
    </section>
  </section>
</main>
`;
